### PR TITLE
feat(engine): added github workflows scan

### PR DESCRIPTION
**Pull request by freitasmillena**
**Proposed Changes**
- Added the scan in the analyzer
- Created cicd/github folder 
- Added CICD to console.go, filesystem.go
- Added tests
- Updated platforms.md documentation to include CICD 
- Added Unpinned Actions Full Length Commit SHA query and tests

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: pkg/analyzer/analyzer.go [hunk 1]
```diff
@@ -633,7 +662,7 @@ func shouldConsiderGitIgnoreFile(path, gitIgnore string, excludeGitIgnoreFile bo
 	gitIgnorePath := filepath.ToSlash(filepath.Join(path, gitIgnore))
 	_, err := os.Stat(gitIgnorePath)
 
-	if !excludeGitIgnoreFile && err == nil {
+	if !excludeGitIgnoreFile && err == nil && gitIgnore != "" {
```
Comment (pereiramarco011) [inline pkg/analyzer/analyzer.go:665]: This change should probably be on a different PR as it has nothing to do with the added platform, do you agree?

File: pkg/analyzer/analyzer.go [hunk 2]
```diff
@@ -390,10 +391,38 @@ func (a *analyzerInfo) worker(results, unwanted chan<- string, locCount chan<- i
 			results <- ansible
 			locCount <- linesCount
 		}
-	// It could be Ansible, Buildah, CloudFormation, Crossplane, or OpenAPI
+	// It could be Ansible, Buildah, CloudFormation, Crossplane, OpenAPI or CICD
 	case yaml, yml, json, sh:
-		a.checkContent(results, unwanted, locCount, linesCount, ext)
+		if isGithubWorkflowfile(a.filePath) {
```
Comment (pereiramarco011) [inline pkg/analyzer/analyzer.go:396]: since this function will read the file and check regexes for the content, the call to it should be inside checkContent function. as the file is already read there.

**Review by pereiramarco011 — CHANGES_REQUESTED** [review]
Left a couple of comments, tell me what you think 😄 

Keep up the good work 👍

**Review by pereiramarco011 — DISMISSED** [review]
LGTM 😄

**Review by gabriel-cx — APPROVED** [review]
